src/Microsoft.AspNetCore.Mvc.ViewFeatures/IViewComponentHelper.cs

```diff
@@ -16,7 +16,14 @@ public interface IViewComponentHelper
         /// Invokes a view component with the specified <paramref name="name"/>.
         /// </summary>
         /// <param name="name">The name of the view component.</param>
-        /// <param name="arguments">Arguments to be passed to the invoked view component method.</param>
+        /// <param name="arguments">
+        /// Arguments to be passed to the invoked view component method.
+        /// <para>
+        /// The argument can either be an <see cref="System.Collections.Generic.IDictionary{string, object}"/>
+        /// instance, or an anonymous object containing names and values of the arguments for the invoked view
+        /// component method.
```

Comment:
Why invent new words? That is, use something more like the wording for `htmlAttributes` parameters in `IHtmlHelper` and `IHtmlGenerator`:
``` c#
/// An <see cref="object"/> containing arguments to be passed to the invoked view component method.
/// Alternatively, an <see cref="IDictionary{string, object}"/> instance containing the invocation arguments.
```

Note the `IDictionary<TK, TV>` case should come second since we expect it to be used less often i.e. only in extra-dynamic cases.